lib/resources/submissions.js

```diff
@@ -81,9 +83,10 @@ module.exports = (service, endpoint) => {
     // there is no body content, and the HTTP spec requires that HEAD should
     // return exactly what GET would, just without a response body, the only thing
     // we can possibly respond with in either case is no body and a 204 code.
-    service.get(path, endpoint.openRosa(({ Projects }, { params }) =>
+    service.get(path, endpointByTestPath.openRosa(({ Projects, Forms }, { params, auth }) =>
       Projects.getById(params.projectId)
         .then(getOrNotFound)
+        .then(() => (params.xmlFormId ? getForm(auth, params, params.xmlFormId, Forms) : resolve()))
```

Comment:
It's not obvious how this change is related to `401` responses; any pointers?

Reply:
I do think it'd be nice to separate it out if that's not too difficult. I was trying to think through the implications of the change. I don't even remember how this endpoint is used. Is there maybe a reason why it needs to be ultra-performant and shouldn't be doing as many checks? Though I guess if it's checking the project, it should probably check the form as well. 🤔 Anyway, I feel like discussing in a separate PR could be nice!